core-services/src/main/java/org/zalando/nakadi/service/StorageService.java

```diff
@@ -24,75 +23,57 @@
 
 import javax.annotation.PostConstruct;
-import javax.annotation.PreDestroy;
 import java.io.IOException;
 import java.util.List;
 import java.util.Optional;
-import java.util.concurrent.Executors;
-import java.util.concurrent.ScheduledExecutorService;
-import java.util.concurrent.TimeUnit;
 import java.util.function.Function;
 
-@Service
+@Service("storageService")
 public class StorageService {
 
-    public static final String ZK_TIMELINES_DEFAULT_STORAGE = "/nakadi/timelines/default_storage";
     private static final Logger LOG = LoggerFactory.getLogger(StorageService.class);
     private final ObjectMapper objectMapper;
     private final StorageDbRepository storageDbRepository;
-    private final DefaultStorage defaultStorage;
-    private final CuratorFramework curator;
     private final FeatureToggleService featureToggleService;
     private final NakadiAuditLogPublisher auditLogPublisher;
-    private final ScheduledExecutorService zkNotificationThread;
+    private final TransactionTemplate transactionTemplate;
+    private final Environment environment;
 
     @Autowired
     public StorageService(final ObjectMapper objectMapper,
                           final StorageDbRepository storageDbRepository,
-                          @Qualifier("default_storage") final DefaultStorage defaultStorage,
-                          final ZooKeeperHolder zooKeeperHolder,
                           final FeatureToggleService featureToggleService,
-                          final NakadiAuditLogPublisher auditLogPublisher) {
+                          final NakadiAuditLogPublisher auditLogPublisher,
+                          final TransactionTemplate transactionTemplate,
+                          final Environment environment) {
         this.objectMapper = objectMapper;
         this.storageDbRepository = storageDbRepository;
-        this.defaultStorage = defaultStorage;
-        this.curator = zooKeeperHolder.get();
         this.featureToggleService = featureToggleService;
         this.auditLogPublisher = auditLogPublisher;
-        this.zkNotificationThread = Executors.newSingleThreadScheduledExecutor();
+        this.transactionTemplate = transactionTemplate;
+        this.environment = environment;
     }
 
-    private void refreshDefaultStorageAndCreateWatcher() {
-        final byte[] defaultStorageId;
-        try {
-            defaultStorageId = curator.getData()
-                    .usingWatcher(
-                            (Watcher) evt -> zkNotificationThread.submit(this::refreshDefaultStorageAndCreateWatcher))
-                    .forPath(ZK_TIMELINES_DEFAULT_STORAGE);
-        } catch (Exception ex) {
-            LOG.error("Failed to refresh default storage id, updates may be blocked", ex);
-            zkNotificationThread.schedule(this::refreshDefaultStorageAndCreateWatcher, 10, TimeUnit.SECONDS);
+    @PostConstruct
+    public void initializeDefaultStorage() {
```

Comment:
Imagine the production system switched default storage to `storage-002` after some time. 
A new instance that start with the `nakadi.timelines.storage.default=storage-001` will try to create this default storage.
This is fine as long as the `storage-001` is not removed from the system in the first place; and DB actually checks for unique ID. 
But given that there is an option to delete a storage, I would prefer to check if there is a default storage before creating a new one.

Reply:
@thomasabraham 
The check is right after your comment (line 62). 
I removed the comment after previous review. This method is a crutch, and should be removed, as storages is a dynamic information and should be maintained in the database. This method should not exist at all, but it requires much more changes.

Reply:
Ahh, I missed it :+1: